apps/jubmoji-quest/prisma/schema.prisma

```diff
@@ -11,5 +11,63 @@ datasource db {
 }
 
 model Card {
-  id    Int     @id @default(autoincrement())
+  id          Int     @id @default(autoincrement())
+  index       Int     @unique // unique hardcoded index, separate from the autogenerated id
+  name        String
+  description String
+  owner       String  // the owner of a card is set by whoever currently holds the card
+
+  // Relationships
+  prerequisitesFor Quest[] @relation("PrerequisiteCards")
+  collectsFor      Quest[] @relation("CollectionCards")
+}
+
+model Quest {
+  id          Int      @id @default(autoincrement())
+  name        String
+  description String
+  tags        CardTag[]?
+  startTime   DateTime?
+  endTime     DateTime?
+  proofType   ProofType
+  proofParams Json    // params for the specific proof type
+
+  // Relationships
+  prerequisiteCards Card[] @relation("PrerequisiteCards") // cards needed to see the quest
+  collectionCards   Card[] @relation("CollectionCards")   // cards needed to complete the quest
+  powers            Power[] // powers granted upon completion
+}
+
+model Power {
+  id          Int     @id @default(autoincrement())
+  name        String
+  description String
+  powerType   PowerType
+  powerParams Json   // params for the specific power type
```

Comment:
How would this interact with the quest expiry time? Can you use a power if it is not expired but the quest is?

Reply:
In my mind definitely, they have different expiries (in some cases, powers would only be unlocked upon quest expiry).

Other examples:
  - all the stuff we cook up that's Zuconnect specific (TG posting) would end at the end of Zuconnect
  - same for Devconnect
  - Time limited prizes where we only have a booth open for 1 hour to claim something

Reply:
@AndrewCLu what do you think? I would like to include one in case it's something we need to display. We can always not use the expiry field if it doesn't become relevant

Reply:
Yeah I agree, I'll add a startTime and endTime, just as for quests - we can ignore them if not needed